File: Task3/Sample.cs
namespace Task3
{
    enum TypeChimicalElement
    {
        Lithium,
        Cobalt,
        Nickel,
        Iron,
        Gold,
        Mercury
    }

    class Sample
    {
        public int number;
        public ChimicalElement[] chimicalElements = new ChimicalElement[6];

        public Sample()
        {
            for (int i = 0; i < chimicalElements.Length; i++)
            {
                chimicalElements[i].Name = "";
                chimicalElements[i].Procent = 0;
            }
        }

        public struct ChimicalElement
        {
            public string Name { get; set; }
            private int procent;

            public int Procent
            {
                set
                {
                    if (value >= 0 && value <= 100)
                        procent = value;
                }
                get => procent;
            }
        }
    }
}

File: Task_3/Sample.cpp
#include <cstdint>
#include <string>

struct ChimicalElement
{
public:
	std::string Name;

	int16_t GetProcent()
	{
		return procent;
	}
	void SetProcent(int16_t value)
	{
		if (value >= 0 && value <= 100)
			procent = value;
	}
private:
	int16_t procent = 0;
};

class Sample
{
public:
	int32_t number = 0;
	ChimicalElement *chimicalElements = new ChimicalElement[6];

	Sample()
	{
		for (int i = 0; i < 6; i++)
		{
			chimicalElements[i].Name = "";
		}
	}
};